frontend/src/components/TimeSheet/WeekCalendar/WeekCalendar.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2020 Lunatech S.A.S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './WeekCalendar.less';
+import {totalHoursPerDay, isPublicHoliday} from '../../../utils/momentUtils';
+import WeekNavigationPanel from './WeekNavigationPanel';
+import {getMaximumHoursPerDay} from '../../../utils/configUtils';
+import {
+  IsDayWithoutAnyEntries,
+  IsDisabled,
+  UserEventEntryData,
+  UserTimeEntryData
+} from '../CalendarUtils';
+import WeekDayCard from './WeekDayCard';
+
+const WeekCalendar = (props) => {
+  const {
+    onClickButton,
+    onClickCard,
+    onClickEntryCard,
+    onPanelChange,
+    onDateChange,
+    publicHolidays,
+    firstDay,
+    userEvents,
+    timeEntriesData,
+    disabledWeekEnd
+  } = props;
+
+  const computeWeeklyTimeEntriesByDays = () => {
+    const daysOfWeek = [...Array(7).keys()].map(i => firstDay.clone().add(i, 'day'));
+    return daysOfWeek.map(dateAsMoment => {
+      return {
+        date: dateAsMoment,
+        timeEntriesForADay: UserTimeEntryData(dateAsMoment, timeEntriesData)
+      };
+    });
+  };
+
+  const weeklyTimeEntriesByDays = computeWeeklyTimeEntriesByDays();
+
+  return (
+    <div id="tk_WeekCalendar">
+      <WeekNavigationPanel
+        firstDay={firstDay}
+        onDateChange={onDateChange}
+        onPanelChange={onPanelChange}
+      />
+      <div id="tk_WeekCalendar_Body">
+        { weeklyTimeEntriesByDays.map(timeEntries => {
+
+          const dateAsMoment = timeEntries.date;
+          const timeEntriesForADay = timeEntries.timeEntriesForADay.data;
+          const eventEntries = UserEventEntryData(userEvents, dateAsMoment);
+
+          const hoursCompleted = totalHoursPerDay(userEvents, dateAsMoment, timeEntriesForADay) >= getMaximumHoursPerDay();
+          const isItPublicHoliday = isPublicHoliday(dateAsMoment, publicHolidays);
```

Comment:
See previous comment about 'getBusinessDays'

Reply:
Yes it is good to use business days but publicHolidays are hard coded at the moment in backend so avoid using isPublicHolidays function will check against the data coming from the backend. But yes weekend field can be avoided.
This can be updated later when we update getBusinessDays function to check against backend publicHolidays data